db/crud.go

```diff
@@ -2350,8 +2432,10 @@ func (col *DatabaseCollectionWithUser) getChannelsAndAccess(ctx context.Context,
 				err = base.HTTPErrorf(500, "JS sync function timed out")
 			} else {
 				err = base.HTTPErrorf(500, "Exception in JS sync function")
-				col.collectionStats.SyncFunctionExceptionCount.Add(1)
-				col.dbStats().Database().SyncFunctionExceptionCount.Add(1)
+				if !dryRun {
```

Comment:
I don't think we want to log a warning (as above) in the case of a dry run either.  The Info and Debug logging above should probably also include some indication that they are associated with a dry run.

Reply:
We're still logging a warning here (line 2441) in the dry run case, which we don't want to do.  You can use the warn_count stat to write a unit test to make sure the diagnostic endpoints aren't triggering these warnings.